teacher_student_dynamics/data_modules/hidden_manifold.py

```diff
@@ -37,7 +39,17 @@ def __init__(
 
         self._activation_name = activation
         self._activation = self._get_activation_function()
-        self._feature_matrix = feature_matrix # DxN matrix
+
+        if rotation_matrix is not None:
+            self._unrotated_feature_matrix = feature_matrix.to(torch.float32)
+            if zero_matrix is not None:
+                feature_matrix = torch.vstack((feature_matrix, zero_matrix)).to(
```

Comment:
Should this line not use `self.unrotated_feature_matrix`?

Reply:
Generally what is being used by the data module is the feature matrix. We only define the unrotated feature matrix because we want to be able to back it out for use in the projections for the teacher. So we don't want unrotated feature matrix to change when feature matrix changes. Does that make sense?

Reply:
So co-opting the pass by reference vs pass by value distinction, it's more like pass by value i.e. when we assign feature matrix to the self._unrotated_feature_matrix variable it's essentially creating a copy of that element rather than a pointer/reference to whatever is being stored in feature_matrix.